stdlib/public/Platform/CMakeLists.txt

```diff
@@ -68,11 +64,34 @@ foreach(sdk ${SWIFT_SDKS})
               "-DGLIBC_INCLUDE_PATH=${GLIBC_INCLUDE_PATH}"
               "-DGLIBC_ARCH_INCLUDE_PATH=${GLIBC_ARCH_INCLUDE_PATH}")
 
+      list(APPEND glibc_modulemap_target_list ${glibc_modulemap_target})
+
+      # If this SDK is a target for a non-native host, install a native modulemap 
+      # without a sysroot prefix.
+      if(NOT "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_PATH}" STREQUAL "/")
+
+        set(glibc_native_modulemap_out "${module_dir}/native-substitutes/glibc.modulemap")
+        handle_gyb_source_single(glibc_modulemap_native_target
+            SOURCE "${glibc_modulemap_source}"
+            OUTPUT "${glibc_native_modulemap_out}"
+            FLAGS
+                "-DCMAKE_SDK=${sdk}"
+                "-DGLIBC_INCLUDE_PATH=${GLIBC_NATIVE_INCLUDE_PATH}"
+                "-DGLIBC_ARCH_INCLUDE_PATH=${GLIBC_NATIVE_ARCH_INCLUDE_PATH}")
+
+        list(APPEND glibc_modulemap_target_list ${glibc_modulemap_native_target})
+        set(glibc_modulemap_out ${glibc_native_modulemap_out})
+      endif()
+
+      # FIXME: When SDK is a cross-compile target (SDK != Host), the generated
+      #        modulemap will be relative to the Host, with hardcoded paths.
+      #        It is not relocatable to the target platform itself.
+      #        This only affects ANDROID right now, but could affect cross-compiled LINUX targets
+        
       swift_install_in_component(sdk-overlay
           FILES "${glibc_modulemap_out}"
```

Comment:
We set `glibc_modulemap_out` to `glibc_sysroot_relative_modulemap_out` (line 84), so it'll install the sysroot-relative one instead. The prefixed one then just becomes a build artefact.